SGoopas/Assets/Scripts/Game/2D/2D Player States/AirDashRight2D.cs

```diff
@@ -0,0 +1,123 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+namespace PlayerStates
+{
+    public class AirDashRight2D : Base2DState
+    {
+        private float afterImageTime = 0.1f;
+        private float timeEllapsed;
+        private GameObject afterimage;
+        private Vector3 direction;
+        private IEnumerator coroutine;
+        private Rigidbody2D rigidbody;
+        private float origGrav;
+        public AirDashRight2D(BasePlayerState previousState) : base(previousState)
+        {
+            direction = new Vector3();
+            PseudoConstructor();
+        }
+        public AirDashRight2D(GameObject player, MasterPlayerStateMachine playerStateMachine, Transform groundCheck) : base(player, playerStateMachine, groundCheck)
+        {
+            direction = new Vector3();
+            PseudoConstructor();
+        }
+
+        public AirDashRight2D(BasePlayerState previousState, Vector3 direction) : base(previousState)
+        {
+            this.direction = direction;
+            PseudoConstructor();
+        }
+        public AirDashRight2D(GameObject player, MasterPlayerStateMachine playerStateMachine, Transform groundCheck, Vector3 direction) : base(player, playerStateMachine, groundCheck)
```

Comment:
I feel like you no longer need direction in the constructor. Every case you construct it here you grab it from DashVector, why not just do that internally? You can even grab it from the previous state in the constructor if that's what you need.

Reply:
DashVector is a calculated property. Perhaps i should rename it to InitialDashVector or something

I wanted the dashing states to be able to work with any direction input (for instance, grounded dashes may always input straight forward), it just happens that all of the states that causes you to dash use DashVector to determine the direction.
I might actually change the grounded states to always sent forward, instead of toward enemies

Reply:
Hmm... How about compromise and make the default constructor use DashVector. That way arbitrary vectors are still supported, but you don't need to use them by default.

Reply:
done